hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientManager.java

```diff
@@ -150,11 +201,16 @@ public void releaseClient(XceiverClientSpi client, boolean invalidateClient) {
     }
   }
 
-  private XceiverClientSpi getClient(Pipeline pipeline)
+  private XceiverClientSpi getClient(Pipeline pipeline, boolean forRead)
       throws IOException {
     HddsProtos.ReplicationType type = pipeline.getType();
     try {
+      // create different client for read different pipeline node based on
+      // network topology
       String key = pipeline.getId().getId().toString() + type;
```

Comment:
Can we wrap this logic in a help function like getPipelineKey()?

Reply:
Sure.